x/leverage/keeper/msg_server.go

```diff
@@ -98,14 +99,21 @@ func (s msgServer) MaxWithdraw(
 	if err != nil {
 		return nil, err
 	}
+	if types.HasUTokenPrefix(msg.Denom) {
+		return nil, types.ErrUToken
+	}
+	if _, err = s.keeper.GetTokenSettings(ctx, msg.Denom); err != nil {
+		return nil, err
+	}
 
 	uToken, err := s.keeper.maxWithdraw(ctx, supplierAddr, msg.Denom)
 	if err != nil {
 		return nil, err
 	}
 
 	if uToken.IsZero() {
-		return nil, types.ErrMaxWithdrawZero
+		zeroCoin := sdkutil.ZeroCoin(msg.Denom)
+		return &types.MsgMaxWithdrawResponse{Withdrawn: uToken, Received: zeroCoin}, nil
```

Comment:
`uToken` here is `sdk.Coin{}` still. do we want to give it a denom?

Reply:
it has a denom, and it's not an "empty" coin, it has zero amount.